fem/prestriction.cpp

```diff
@@ -858,11 +882,94 @@ void ParNCL2FaceRestriction::FillI(SparseMatrix &mat,
    MFEM_ABORT("Not yet implemented.");
 }
 
-void ParNCL2FaceRestriction::FillJAndData(const Vector &ea_data,
+void ParNCL2FaceRestriction::FillJAndData(const Vector &fea_data,
                                           SparseMatrix &mat,
                                           const bool keep_nbr_block) const
 {
-   MFEM_ABORT("Not yet implemented.");
+   if (keep_nbr_block)
+   {
+      return NCL2FaceRestriction::FillJAndData(fea_data, mat, keep_nbr_block);
+   }
+   const int nface_dofs = face_dofs;
+   const int Ndofs = ndofs;
+   auto d_indices1 = scatter_indices1.Read();
+   auto d_indices2 = scatter_indices2.Read();
+   auto I = mat.ReadWriteI();
+   auto mat_fea = Reshape(fea_data.Read(), nface_dofs, nface_dofs, 2, nf);
+   auto J = mat.WriteJ();
+   auto Data = mat.WriteData();
+   auto interp_config_ptr = interpolations.GetFaceInterpConfig().Read();
+   auto interpolators = interpolations.GetInterpolators().Read();
+   const int nc_size = interpolations.GetNumInterpolators();
+   auto d_interp = Reshape(interpolators, nface_dofs, nface_dofs, nc_size);
+   mfem::forall(nf*nface_dofs, [=] MFEM_HOST_DEVICE (int fdof)
+   {
+      const int f  = fdof/nface_dofs;
+      const InterpConfig conf = interp_config_ptr[f];
+      const int master_side = conf.master_side;
+      const int interp_index = conf.index;
+      const int iF = fdof%nface_dofs;
+      const int iE1 = d_indices1[f*nface_dofs+iF];
+      if (iE1 < Ndofs)
+      {
+         const int offset1 = AddNnz(iE1,I,nface_dofs);
+         for (int jF = 0; jF < nface_dofs; jF++)
+         {
+            const int jE2 = d_indices2[f*nface_dofs+jF];
+            J[offset1+jF] = jE2;
+            double val2 = 0.0;
```

Comment:
There are several places, like this, where `double` needs to be replaced by `real_t`.

Reply:
@YohannDudouit, can you make this ☝️ change?

Reply:
done.